node_modules/col-analytics/lib/caliper.js

```diff
@@ -58,19 +63,185 @@ var init = module.exports.init = function() {
  * @return {void}
  */
 var track = module.exports.track = function(user, event, metadata) {
-  try {
-    getCourseIfMissing(user, function(err, course) {
-      if (err) {
-        log.error({'user': user}, 'Could not retrieve course for user');
+  getCourseIfMissing(user, function(err, course) {
+    if (err) {
+      log.error({'user': user}, 'Could not retrieve course for user');
+    }
+
+    try {
+      var caliperEvent = buildEvent(user, course, event, metadata);
+      if (caliperEvent) {
+        var parsedEvent = JSON.parse(Caliper.Clients.ClientUtils.stringify(caliperEvent));
+        log.info({'event': parsedEvent}, 'Built Caliper event');
+
+        // TODO send event via sensor
       }
+    } catch (caliperError) {
+      log.error(caliperError);
+    }
+  });
+};
+
+/**
+ * Build a Caliper event from generic event data
+ *
+ * @param  {User}           user              The user associated with the event
+ * @param  {Course}         course            The course associated with the event
+ * @param  {String}         eventDescription  A string descrining the event type
+ * @param  {Object}         [metadata]        Optional event metadata
+ * @return {Object}                           The built Caliper event
+ */
+var buildEvent = function(user, course, eventDescription, metadata) {
+  var actor = buildPerson(user, course);
+  var courseSite = buildOrganization(course);
+  var membership = buildMembership(user, actor, courseSite);
+
+  var action = null;
+  var eventType = null;
+  var object = null;
 
-      // TODO create event based on user + course + metadata and send via sensor
+  switch (eventDescription) {
+    case 'Launch Asset Library':
+      action = Caliper.Actions.navigatedTo;
+      eventType = Caliper.Events.NavigationEvent;
+      object = buildSoftwareApplication('assetlibrary', 'Asset Library');
+      break;
+    case 'Launch Engagement Index':
+      action = Caliper.Actions.navigatedTo;
+      eventType = Caliper.Events.NavigationEvent;
+      object = buildSoftwareApplication('engagementindex', 'Engagement Index');
+      break;
+    case 'Launch Impact Studio':
+      action = Caliper.Actions.navigatedTo;
+      eventType = Caliper.Events.NavigationEvent;
+      object = buildSoftwareApplication('dashboard', 'Impact Studio');
+      break;
+    case 'Launch Whiteboards':
+      action = Caliper.Actions.navigatedTo;
+      eventType = Caliper.Events.NavigationEvent;
+      object = buildSoftwareApplication('whiteboards', 'Whiteboards');
+      break;
+    // If event description is not matched, exit early
+    default:
+      return null;
+  }
+
+  return eventFactory().create(eventType, {
```

Comment:
@johncrossman As more events get added, `action` and `eventType` will vary. I thought about doing the switch in stages, but it seems more readable to just list them in a single mapping.